src/elements/netspeak-search.tsx

```diff
@@ -416,36 +448,51 @@ export class NetspeakSearch extends React.PureComponent<Props, State> {
 	render(): JSX.Element {
 		const l = createLocalizer(this.props, locales);
 		const { warnings, errors } = this._splitProblems();
+		if (this.props.refreshSearch) {
+			this._setQuery("", true);
+			this._setQuery(this.state.query, true); //done to reset normalized query
+			this._queryPhrases(this.state.query);
+			this.props.onSearchRefreshed();
+		}
+
+		if (this.props.showExperimental && this.props.forcedExperimentalQuery !== this.state.query) {
+			this._setQuery(this.props.forcedExperimentalQuery, false);
+		}
```

Comment:
You are never ever allowed to change state during `render`.

This is what React told me:
![image](https://user-images.githubusercontent.com/20878432/134361776-0bcfaad2-66e4-43b3-bfd6-4d44beddf7c6.png)